src/frontend/hooks/goose-form.js

```diff
@@ -81,13 +81,25 @@ const useGooseForm = () => {
 
   const saveForm = useCallback(async () => {
     // TODO: Validation checks here.
-
     // Send data to server:
+
     try {
+      // files added 
       const files = images;
-      let imageUrls = [];
-
-      if (files.length > 0) {
+  
+      // all files uploaded 
+      let imageUrls2 = [];
+      
+      if (files.length > 0 || imageUrls.length > 0) {
+          // when no new files are added
+         if (files.length <= 0){
+          const imgArr = imageUrls.map((url) => url.imgDir)
+          imageUrls2 = imgArr
+          setImageUrls(imageUrls2)
+         }
```

Comment:
Is this redundant (meaning we can remove it)? Because regardless if we are adding files or not, updating the state of `imagesUrls` using a map of `imageUrls` is basically setting it equal to itself, except in this case the structure of the state is actually being altered (this code makes `imageUrls` an array of strings, rather than an array of objects, which is what it was originally (see line 29 of this file) - although the state doesn't matter too much since it gets overriden once you open the form again, this should be avoided)